nerfstudio/data/datamanagers/full_images_datamanager.py

```diff
@@ -172,14 +172,31 @@ def cache_images(self, cache_images_option):
             if mask is not None:
                 data["mask"] = mask
 
-            cached_eval.append(data)
+            self.eval_dataset.cameras.fx[idx] = float(K[0, 0])
+            self.eval_dataset.cameras.fy[idx] = float(K[1, 1])
+            self.eval_dataset.cameras.cx[idx] = float(K[0, 2])
+            self.eval_dataset.cameras.cy[idx] = float(K[1, 2])
+            self.eval_dataset.cameras.width[idx] = image.shape[1]
+            self.eval_dataset.cameras.height[idx] = image.shape[0]
+            return data
 
-            self.eval_dataset.cameras.fx[i] = float(K[0, 0])
-            self.eval_dataset.cameras.fy[i] = float(K[1, 1])
-            self.eval_dataset.cameras.cx[i] = float(K[0, 2])
-            self.eval_dataset.cameras.cy[i] = float(K[1, 2])
-            self.eval_dataset.cameras.width[i] = image.shape[1]
-            self.eval_dataset.cameras.height[i] = image.shape[0]
+        result_list = []
+        CONSOLE.log("Caching / undistorting train images")
+        with ThreadPoolExecutor(max_workers=multiprocessing.cpu_count()) as executor:
```

Comment:
If you're using a thread pool I don't think `multiprocessing.cpu_count()` makes sense here; because of the [GIL](https://wiki.python.org/moin/GlobalInterpreterLock) there's not really any relationship between the optimal number of threads and the number of CPUs.

Since OpenCV will use multiple cores under-the-hood anyways, I think we could just set the number of threads to a high-enough constant values that allows us to parallelize I/O operations (reading images via `.get_data()`) and undistortion.

Reply:
The ThreadPoolExecutor's default num_workers is max_workers = min(32, (os.cpu_count() or 1) + 4). I think this could be reasonable and I just leave the init num empty.